playout/libretime_playout/main.py

```diff
@@ -144,18 +92,21 @@ def cli(log_level: str, log_filepath: Optional[Path], config_filepath: Optional[
             logger.exception(exception)
             time.sleep(10)
 
-    telnet_lock = Lock()
-
-    liquidsoap_host = config.playout.liquidsoap_host
-    liquidsoap_port = config.playout.liquidsoap_port
+    liq_client = LiquidsoapClient(
+        host=config.playout.liquidsoap_host,
+        port=config.playout.liquidsoap_port,
+    )
 
-    liquidsoap_startup_test(telnet_lock, liquidsoap_host, liquidsoap_port)
+    logger.debug("Checking if Liquidsoap is running")
+    liq_version = liq_client.wait_for_version()
+    if not LIQUIDSOAP_MIN_VERSION <= liq_version:
+        raise Exception(f"Invalid liquidsoap version {liq_version}")
```

Comment:
We do not loop indefinitely on invalid version or connection anymore. We fail hard and let the service manager restart the service if wanted.

More generally, I think we should aim for a fast and disposable app that can catch up its previous stats in case of restart. Catching up its state would mean without having to reset everything as it actually does, but simply read the current state and apply a diff update if necessary.

Reply:
I agree with that - it also aligns with HA practices where you cannot guarantee any single instance's state over a period of time